frontend/features/page/sections/ContactsSection/ContactsSection.tsx

```diff
@@ -0,0 +1,675 @@
+        state.validationErrors.name = "";
+        if (
+          state.values.email == "" ||
+          !state.values.email ||
+          state.values.email?.replace(/\s/g, "").length == 0
+        ) {
+          state.validationErrors.email = "Please insert your email";
+          error = true;
+        } else {
+          if (!regex.test(state.values.email)) {
+            state.validationErrors.email = "Please insert a valid email";
+            error = true;
+          } else {
+            state.validationErrors.email = "";
+            if (state.values.message == "" || !state.values.message) {
+              state.validationErrors.message = "Please insert a message";
+              error = true;
+            } else {
+              state.validationErrors.message = "";
+            }
+          }
+        }
+      }
+
+      event?.preventDefault();
+
+      if (error) {
+        return {
+          ...state,
+          status: FormStatus.Idle,
+        };
+      }
+
+      return {
+        ...state,
+        status: FormStatus.Submitting,
+      };
+    }
+
+    case FormActionType.Success:
+      return {
+        // Display thanks page
+        ...state,
+        status: FormStatus.Submitted,
+      };
+
+    case FormActionType.Failed:
+      return {
+        // Resta in idle, setta errore e resetta i campi?
+        ...state,
+        values: {
+          name: "",
+          email: "",
+          message: "",
+        },
+        submitError: "Error while sending your message",
+        status: FormStatus.Idle,
+      };
+
+    default: {
+      assertNever(action);
+    }
+  }
+}
+
+export function ContactsForm() {
+  const [state, dispatch] = React.useReducer(reducer, blankForm);
+  React.useEffect(() => {
+    if (state.status === FormStatus.Submitting) {
+      async function sendMessage() {
+        const input: CreateFormMessageInput = {
+          data: {
+            name: state.values.name,
+            email: state.values.email,
+            message: state.values.message,
+          },
+        };
+        const response = await fetchGraphQL<
+          InsertFormMessageMutation,
+          InsertFormMessageMutationVariables
+        >(InsertFormMessage, { input });
+
+        if (response.createFormMessage == null) {
+          dispatch({ type: FormActionType.Failed });
+        } else {
+          dispatch({ type: FormActionType.Success });
+        }
+      }
+      sendMessage();
+    }
+  }, [state.status, state.values]);
+
+  return (
+    <Box
+      flexWrap="wrap"
+      fontFamily="Roboto Mono"
+      id="contact"
+      mt={{
+        base: "20",
+        md: "0",
+      }}
+      mr={{
+        base: "-2.5",
+        md: "0",
+      }}
+      display="flex"
+      flexGrow={1}
+      pb="20"
+      flexDir={{
+        base: "column",
+        sm: "row",
+        md: "column",
+        lg: "row",
+      }}>
+      <Box
+        display={state.status == FormStatus.Submitted ? "inline-block" : "none"}
```

Comment:
Always use strict comparison `===`. The only place where it makes sense to use `==` is when you a checking for null or undefined with `== null`.